eng/cibuild.sh

```diff
@@ -35,9 +35,38 @@ if [ "$__osname" == "Linux" ]; then
             source scl_source enable python27 devtoolset-2
         fi
     fi
+
+    # We are using old (2019) centos image in the CI with old cmake (2.8).
+    # Upgrading to 2021 centos image was failing SOS tests which rely on
+    # lldb REPL and ptrace etc. e.g. from test attachment logs:
+    #
+    #             00:00.136: error: process launch failed: 'A' packet returned an error: 8
+    #             00:00.136:
+    #             00:00.136: <END_COMMAND_ERROR>
+    #System.Exception: 'process launch -s' FAILED
```

Comment:
@mikem8361, @hoyosjs, for some reason only 2019 centos7 image is able to run LLDB tests, changing https://github.com/dotnet/diagnostics/blob/9dc72da9618bbb8c57b902d5e3721ebdfa47b681/diagnostics.yml#L129 to 2021 one:

```yml
dockerImage: mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7-20210714125435-9b5bbc2
```
was failing lldb tests with that error.

Reply:
Yup, this is a large part of the reason this section is so stale - no one has actually had the time to go an fix the container. I've seen this error locally when using lldb attach in some scenarios is SYS_PTRACE is not available, but I doubt that's the problem here (same kernel used between the two images, and doubt they'll start them with different seccomp settings).

Reply:
In case of a container, the kernel used is from host machine so that's same as you have mentioned.
I think it could be the difference in how these containers are built. There has been many changes in dotnet docker build tool (which also builds prereq images).